plugins/scanners/fossid/src/main/kotlin/events/UploadArchiveHandler.kt

```diff
@@ -125,4 +131,19 @@ class UploadArchiveHandler(
         service.removeUploadedContent(config.user.value, config.apiKey.value, scanCode)
             .checkResponse("remove previously uploaded content 2", false)
     }
+
+    internal fun deleteExcludedFiles(path: File, includes: Includes?, excludes: Excludes?) {
+        path.walkBottomUp()
+            .filter { it != path }
+            .forEach { file ->
+                val relativePath = file.relativeTo(path).toString()
+                if (shouldDeleteFile(relativePath, includes, excludes)) {
+                    file.delete()
+                }
+            }
+    }
+
+    private fun shouldDeleteFile(relativePath: String, includes: Includes?, excludes: Excludes?): Boolean =
+        (includes != null && !includes.isPathIncluded(relativePath)) ||
```

Comment:
This can be combined to `includes?.isPathIncluded(relativePath)) == false`. Similar below.